clar2wasm/src/words/sequences.rs

```diff
@@ -270,6 +272,28 @@ impl ComplexWord for Append {
             .ok_or_else(|| GeneratorError::TypeError("append result must be typed".to_string()))?
             .clone();
 
+        let list = args.get_expr(0)?;
+        let elem = args.get_expr(1)?;
+
+        // WORKAROUND: setting correct types for arguments
+        match &ty {
+            TypeSignature::SequenceType(SequenceSubtype::ListType(ltd)) => {
+                generator.set_expr_type(
+                    list,
+                    #[allow(clippy::unwrap_used)]
+                    ListTypeData::new_list(ltd.get_list_item_type().clone(), ltd.get_max_len() - 1)
+                        .unwrap() // this type is same as expr but smaller -> valid
```

Comment:
use expect() with string instead of comment

Reply:
also spell out "is thus always valid" or similar, instead of ->

Reply:
If I use `expect`, the message should be more of the style "it failed when it shouldn't have" then :/